File: src/types/library.ts
/** Mirrors `backend/schemas/library.py::DocumentSummaryOut`. */

export type DocumentStatus =
  | "transcribing"
  | "uploaded"
  | "processing"
  | "ready"
  | "error";

/** Which section a document belongs to: full study vs assessment-only exam prep. */
export type DocumentMode = "study" | "exam";

/** Where a document came from: an uploaded PDF, or transcribed audio. */
export type SourceType = "pdf" | "audio";

export interface DocumentSummary {
  id: number;
  filename: string;
  subject_id: number;
  subject_name: string;
  subject_bookmarked: boolean;
  /** ISO date (YYYY-MM-DD) or null when no exam date is set. */
  exam_date: string | null;
  status: DocumentStatus;
  /** Human-readable detail for the status (e.g. a transcription wait message). */
  status_detail: string | null;
  source_type: SourceType;
  mode: DocumentMode;
  /** ISO datetime. */
  uploaded_at: string;
  topics_total: number;
  topics_ready: number;
  chapters_total: number;
  /** Chapter lanes set to process (running). */
  chapters_running: number;
}

File: backend/schemas/library.py
"""Schemas for the Library (home screen) document list (Phase 9)."""

from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict

from backend.models.enums import DocumentMode, DocumentStatus


class DocumentSummaryOut(BaseModel):
    """A document as shown in the Library list: metadata + topic-ready progress."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    filename: str
    subject_id: int
    subject_name: str
    subject_bookmarked: bool
    exam_date: date | None
    status: DocumentStatus
    status_detail: str | None
    source_type: str  # "pdf" | "audio"
    mode: DocumentMode
    uploaded_at: datetime
    topics_total: int
    topics_ready: int
    chapters_total: int
    chapters_running: int  # chapter lanes set to process (running)
